src/__swaps__/safe-math/SafeMath.ts

```diff
@@ -0,0 +1,146 @@
+  const integerPart = resultStr.slice(0, -20) || '0';
+  let fractionalPart = resultStr.slice(-20);
+  fractionalPart = fractionalPart.replace(/0+$/, ''); // Remove trailing zeros
+  return fractionalPart.length > 0 ? `${integerPart}.${fractionalPart}` : integerPart;
+};
+
+// Sum function
+export function sum(num1: string, num2: string): string {
+  if (!isNumberString(num1) || !isNumberString(num2)) {
+    throw new Error('Arguments must be a numeric string');
+  }
+  if (isZero(num1)) {
+    return num2;
+  }
+
+  if (isZero(num2)) {
+    return num1;
+  }
+  const [bigInt1, decimalPlaces1] = removeDecimal(num1);
+  const [bigInt2, decimalPlaces2] = removeDecimal(num2);
+  const scaledBigInt1 = scaleUp(bigInt1, decimalPlaces1);
+  const scaledBigInt2 = scaleUp(bigInt2, decimalPlaces2);
+  const result = scaledBigInt1 + scaledBigInt2;
+  return formatResult(result);
+}
+
+// Subtract function
+export function sub(num1: string, num2: string): string {
+  if (!isNumberString(num1) || !isNumberString(num2)) {
+    throw new Error('Arguments must be a numeric string');
+  }
+
+  if (isZero(num2)) {
+    return num1;
+  }
+
+  const [bigInt1, decimalPlaces1] = removeDecimal(num1);
+  const [bigInt2, decimalPlaces2] = removeDecimal(num2);
+  const scaledBigInt1 = scaleUp(bigInt1, decimalPlaces1);
+  const scaledBigInt2 = scaleUp(bigInt2, decimalPlaces2);
+  const result = scaledBigInt1 - scaledBigInt2;
+  return formatResult(result);
+}
+
+// Multiply function
+export function mul(num1: string, num2: string): string {
+  if (!isNumberString(num1) || !isNumberString(num2)) {
+    throw new Error('Arguments must be a numeric string');
+  }
+  if (isZero(num1) || isZero(num2)) {
+    return '0';
+  }
+  const [bigInt1, decimalPlaces1] = removeDecimal(num1);
+  const [bigInt2, decimalPlaces2] = removeDecimal(num2);
+  const scaledBigInt1 = scaleUp(bigInt1, decimalPlaces1);
+  const scaledBigInt2 = scaleUp(bigInt2, decimalPlaces2);
+  const result = (scaledBigInt1 * scaledBigInt2) / BigInt(10) ** BigInt(20);
+  return formatResult(result);
+}
+
+// Divide function
+export function div(num1: string, num2: string): string {
+  if (!isNumberString(num1) || !isNumberString(num2)) {
+    throw new Error('Arguments must be a numeric string');
+  }
+  if (isZero(num2)) {
+    throw new Error('Division by zero');
+  }
+  if (isZero(num1)) {
+    return '0';
+  }
+  const [bigInt1, decimalPlaces1] = removeDecimal(num1);
+  const [bigInt2, decimalPlaces2] = removeDecimal(num2);
+  const scaledBigInt1 = scaleUp(bigInt1, decimalPlaces1);
+  const scaledBigInt2 = scaleUp(bigInt2, decimalPlaces2);
+  const result = (scaledBigInt1 * BigInt(10) ** BigInt(20)) / scaledBigInt2;
+  return formatResult(result);
+}
+
+// Modulus function
+export function mod(num1: string, num2: string): string {
+  if (!isNumberString(num1) || !isNumberString(num2)) {
+    throw new Error('Arguments must be a numeric string');
+  }
+  if (isZero(num1) && isZero(num2)) {
```

Comment:
i think this is too strict, there should be a division by zero error if only num2 (the divisor) is zero. num1 is allowed to be zero